overviews/core/_posts/2012-09-20-futures.md

```diff
@@ -36,6 +36,134 @@ in the case of an application which requires blocking futures, for an underlying
 environment to resize itself if necessary to guarantee progress.
 -->
 
+A typical future look like this:
+
+    val inverseFuture : Future[Matrix] = Future {
+      fatMatrix.inverse() // non-blocking long lasting computation
+    }(executionContext)
+
+
+Or with the more idiomatic
+
+    implicit val ec : ExecutionContext = ...
+    val inverseFuture : Future[Matrix] = Future {
+      fatMatrix.inverse()
+    } // ec is implicitly passed
+
+
+Both code snippets delegate the execution of `fatMatrix.inverse()` to an `ExecutionContext` and embody the result of the computation in `inverseFuture`.
+
+
+## Execution Context   
+
+Future and Promises revolve around [ExecutionContexts](http://www.scala-lang.org/api/current/index.html#scala.concurrent.ExecutionContext), responsible for executing computations.
+
+An `ExecutionContext` is similar to an [Executor](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executor.html): it is free to execute  computations in a new thread, in a pooled thread or in the current thread (although this is discouraged - more on that below).
+
+The `scala.concurrent` package comes out of the box with an `ExecutionContext` implementation, a global static thread pool.
+It is also possible to adapt an `Executor` into an `ExecutionContext`, and you are of course free to implement your own. 
+
+
+### The Global Execution Context
+
+`ExecutionContext.global` is an `ExecutionContext` backed by a [ForkJoinPool](http://docs.oracle.com/javase/tutorial/essential/concurrency/forkjoin.html). It should be sufficient for most situations but requires some care.  
```

Comment:
I'm having second thoughts about mentioning FJ pool here.
The current `ExecutionContext.global` implementation, even though it's unlikely to change,
should be mentioned at most in the ScalaDoc API.
Otherwise, our documentation is too flaky, and it becomes hard to maintain the crossreferences.

This doc should discuss only the high-level usage and meaning of execution contexts,
not the specific implementation.
It is also unclear if we should mention `minThreads`, `numThreads` and `maxThreads` configuration settings,
since they are not mentioned explicitly in the API docs.